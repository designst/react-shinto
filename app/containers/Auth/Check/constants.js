import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
