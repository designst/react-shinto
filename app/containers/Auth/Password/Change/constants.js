import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_CHANGE_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_REQUEST',
);
export const AUTH_PASSWORD_CHANGE_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_SUCCESS',
);
export const AUTH_PASSWORD_CHANGE_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_FAILURE',
);
