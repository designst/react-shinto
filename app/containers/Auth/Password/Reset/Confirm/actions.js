import {
  AUTH_PASSWORD_RESET_CONFIRM_REQUEST,
  AUTH_PASSWORD_RESET_CONFIRM_SUCCESS,
  AUTH_PASSWORD_RESET_CONFIRM_FAILURE,
} from './constants';

export const passwordResetConfirmRequest = payload => ({
  type: AUTH_PASSWORD_RESET_CONFIRM_REQUEST,
  payload,
});

export const passwordResetConfirmSuccess = data => ({
  type: AUTH_PASSWORD_RESET_CONFIRM_SUCCESS,
  data,
});

export const passwordResetConfirmFailure = error => ({
  type: AUTH_PASSWORD_RESET_CONFIRM_FAILURE,
  error,
});
