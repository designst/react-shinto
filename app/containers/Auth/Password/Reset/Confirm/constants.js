import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_CONFIRM_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_REQUEST',
);
export const AUTH_PASSWORD_RESET_CONFIRM_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_SUCCESS',
);
export const AUTH_PASSWORD_RESET_CONFIRM_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_FAILURE',
);
