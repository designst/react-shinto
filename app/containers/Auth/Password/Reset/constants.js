import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_REQUEST',
);
export const AUTH_PASSWORD_RESET_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_SUCCESS',
);
export const AUTH_PASSWORD_RESET_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_FAILURE',
);
