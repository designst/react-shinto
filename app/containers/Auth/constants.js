import createConstant from 'utils/createConstant';

export const AUTH_API_REQUEST = createConstant(__filename, 'AUTH_API_REQUEST');
export const AUTH_API_SUCCESS = createConstant(__filename, 'AUTH_API_SUCCESS');
export const AUTH_API_FAILURE = createConstant(__filename, 'AUTH_API_FAILURE');
