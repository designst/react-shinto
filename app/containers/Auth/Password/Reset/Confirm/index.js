import { compose } from 'redux';
import { connect } from 'react-redux';

import PasswordResetConfirm from 'components/Auth/Password/Reset/Confirm';

import { passwordResetConfirmRequest } from './actions';

const mapStateToProps = (state, { match }) => ({
  initialValues: {
    token: match.params.token,
  },
});

const mapDispatchToProps = {
  onSubmit: passwordResetConfirmRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(PasswordResetConfirm);
