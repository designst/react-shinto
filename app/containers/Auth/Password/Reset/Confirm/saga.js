import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_RESET_CONFIRM_REQUEST } from 'containers/Auth/Password/Reset/Confirm/constants';

import {
  passwordResetConfirmSuccess,
  passwordResetConfirmFailure,
} from 'containers/Auth/Password/Reset/Confirm/actions';

const logger = createLogger(__filename);

const authPasswordResetConfirmUrl = process.env.SHINTO_AUTH_PASSWORD_RESET_CONFIRM_API_ENDPOINT;

export function* authPasswordResetConfirm(action) {
  const { token } = action;

  const apiService = yield getContext('apiService');

  logger('Password Reset Confirm: %s', token);

  try {
    const response = yield call(apiService.post, authPasswordResetConfirmUrl, {
      token,
    });
    yield put(passwordResetConfirmSuccess(response.data));
  } catch (err) {
    yield put(passwordResetConfirmFailure(err));
  }
}

export default function* authPasswordResetConfirmData() {
  yield takeLatest(AUTH_PASSWORD_RESET_CONFIRM_REQUEST, authPasswordResetConfirm);
}
