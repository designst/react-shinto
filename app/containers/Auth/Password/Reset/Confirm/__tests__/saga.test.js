import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import authPasswordResetConfirmData from 'containers/Auth/Password/Reset/Confirm/saga';
import { AUTH_PASSWORD_RESET_CONFIRM_REQUEST } from 'containers/Auth/Password/Reset/Confirm/constants';

describe('authPasswordResetConfirmData', () => {
  let authPasswordResetConfirmGenerator;

  beforeEach(() => {
    authPasswordResetConfirmGenerator = authPasswordResetConfirmData();

    const selectDescriptor = authPasswordResetConfirmGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authPasswordResetConfirmSuccess action if it requests the data successfully', () => {
    const token = 'token';

    return expectSaga(authPasswordResetConfirmData)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/password/reset/confirm'
              ? {
                  data: {
                    token,
                  },
                }
              : {},
        }),
      })
      .dispatch({
        type: AUTH_PASSWORD_RESET_CONFIRM_REQUEST,
        payload: {
          token,
        },
      })
      .withReducer(authReducer)
      .run();
  });

  it('should dispatch the authPasswordResetConfirmFailure action if it requests the data unsuccessfully', () => {
    const token = 'token';

    return expectSaga(authPasswordResetConfirmData)
      .dispatch({
        type: AUTH_PASSWORD_RESET_CONFIRM_REQUEST,
        payload: {
          token,
        },
      })
      .withReducer(authReducer)
      .run();
  });
});
