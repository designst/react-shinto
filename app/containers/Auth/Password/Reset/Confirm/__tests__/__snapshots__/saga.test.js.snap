// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authPasswordResetConfirmData should dispatch the authPasswordResetConfirmFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/reset/confirm/constants/AUTH_PASSWORD_RESET_CONFIRM_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`authPasswordResetConfirmData should dispatch the authPasswordResetConfirmSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/reset/confirm/constants/AUTH_PASSWORD_RESET_CONFIRM_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
